phantazm-zombies-mapdata/src/main/java/com/github/phantazmnetwork/zombies/map/FileSystemMapLoader.java

```diff
@@ -0,0 +1,167 @@
+                MapProcessors.doorInfo())));
+
+        FileUtils.forEachFileMatching(paths.shops, configPredicate, file -> shops.add(ConfigBridges.read(file, codec,
+                MapProcessors.shopInfo())));
+
+        FileUtils.forEachFileMatching(paths.windows, configPredicate, file -> windows.add(ConfigBridges.read(file,
+                codec, MapProcessors.windowInfo())));
+
+        FileUtils.forEachFileMatching(paths.rounds, configPredicate, file -> rounds.add(ConfigBridges.read(file, codec,
+                MapProcessors.roundInfo())));
+
+        FileUtils.forEachFileMatching(paths.spawnrules, configPredicate, file -> spawnrules.add(ConfigBridges.read(file,
+                codec, MapProcessors.spawnruleInfo())));
+
+        FileUtils.forEachFileMatching(paths.spawnpoints, configPredicate, file -> spawnpoints.add(ConfigBridges
+                .read(file, codec, MapProcessors.spawnpointInfo())));
+
+        rounds.sort(Comparator.comparingInt(RoundInfo::round));
+
+        return new MapInfo(mapSettingsInfo, rooms, doors, shops, windows, rounds, spawnrules, spawnpoints);
+    }
+
+    @Override
+    public void save(@NotNull MapInfo data) throws IOException {
+        Path mapDirectory = root.resolve(data.info().id().value());
+        Files.createDirectories(mapDirectory);
+
+        MapSettingsInfo mapSettingsInfo = data.info();
+        ConfigBridges.write(mapDirectory.resolve(mapInfoName), MapProcessors.mapInfo().elementFromData(mapSettingsInfo), codec);
+
+        FolderPaths paths = new FolderPaths(mapDirectory);
+
+        FileUtils.deleteRecursivelyIfExists(paths.rooms);
+        FileUtils.deleteRecursivelyIfExists(paths.doors);
+        FileUtils.deleteRecursivelyIfExists(paths.shops);
+        FileUtils.deleteRecursivelyIfExists(paths.windows);
+        FileUtils.deleteRecursivelyIfExists(paths.rounds);
+        FileUtils.deleteRecursivelyIfExists(paths.spawnrules);
+        FileUtils.deleteRecursivelyIfExists(paths.spawnpoints);
+
+        Files.createDirectories(paths.rooms);
+        Files.createDirectories(paths.doors);
+        Files.createDirectories(paths.shops);
+        Files.createDirectories(paths.windows);
+        Files.createDirectories(paths.rounds);
+        Files.createDirectories(paths.spawnrules);
+        Files.createDirectories(paths.spawnpoints);
+
+        String extension = "." + codec.getPreferredExtension();
+
+        for(RoomInfo room : data.rooms()) {
+            ConfigBridges.write(paths.rooms.resolve(room.id().value() + extension), codec, MapProcessors
+                    .roomInfo(), room);
+        }
+
+        for(DoorInfo door : data.doors()) {
+            ConfigBridges.write(paths.doors.resolve(door.id().value() + extension), MapProcessors
+                    .doorInfo().elementFromData(door), codec);
+        }
+
+        for(ShopInfo shop : data.shops()) {
+            ConfigBridges.write(paths.shops.resolve(getPositionString(shop.triggerLocation()) + "-" + shop.id()
```

Comment:
why name it this way

Reply:
easier to identify where the shop is if the coordinates are in the filename